Test/PurchaseTest.cs

```diff
@@ -29,6 +29,116 @@ public void InvoicePurchase()
+            purchase.Account.BillingInfo = sub.Account.BillingInfo;
+            purchase.Subscriptions.Add(sub);
+
+            var collection = Purchase.Invoice(purchase);
+            int firstRampUnitAmount = sub.RampIntervals[0].UnitAmountInCents;
+
+            Assert.NotNull(collection.ChargeInvoice);
+            Assert.Equal(collection.ChargeInvoice.SubtotalInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.TotalInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.SubtotalBeforeDiscountInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.Adjustments[0].UnitAmountInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.Adjustments[0].TotalInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.Transactions[0].AmountInCents, firstRampUnitAmount);
+            account.Close();
+        }
+
+        [RecurlyFact(TestEnvironment.Type.Integration)]
+        public void PreviewPurchaseWithRamps()
+        {
+            var account = CreateNewAccountWithBillingInfo();
+            var currency = "USD";
+            var plan = CreateNewRampPlan(3);
+            PlansToDeactivateOnDispose.Add(plan);
+
+            var purchase = new Purchase(account.AccountCode, currency);
+            purchase.Account.BillingInfo = account.BillingInfo;
+
+            var sub = new Subscription(plan.PlanCode);
+            purchase.Subscriptions.Add(sub);
+
+            var collection = Purchase.Preview(purchase);
+            int firstRampUnitAmount = plan.RampIntervals[0].Currencies[0].UnitAmountInCents;
+
+            Assert.NotNull(collection.ChargeInvoice);
+            Assert.Equal(collection.ChargeInvoice.SubtotalInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.TotalInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.SubtotalBeforeDiscountInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.Adjustments[0].UnitAmountInCents, firstRampUnitAmount);
+            Assert.Equal(collection.ChargeInvoice.Adjustments[0].TotalInCents, firstRampUnitAmount);
+            account.Close();
+        }
+
+        [RecurlyFact(TestEnvironment.Type.Integration)]
+        public void PendingPurchaseWithRamps()
+        {
+            var currency = "USD";
+            var plan = CreateNewRampPlan(3);
+            PlansToDeactivateOnDispose.Add(plan);
+            var account = CreateNewAccountWithBillingInfo();
+
+            var sub = new Subscription(account, plan, "USD");
+            sub.RampIntervals = GetMockSubscriptionRampIntervals(2);
+
+            var purchase = new Purchase(sub.AccountCode, currency);
+            purchase.Account.BillingInfo = sub.Account.BillingInfo;
+            // Randomly generate an email address
+            purchase.Account.Email = Guid.NewGuid().ToString().Substring(0, 8) + "@test.com";
```

Comment:
nit: any reason not to just "user@example.com"?

Reply:
Just to prevent the test from being flakey.  i guess i wasnt wanting accounts to be out there all with the same email incase something somehow didnt make it to the account.close().  Everything else in the tests are setup to get new random string, for example plan_code or plan name.